diff --git a/dist/node/chunks/config.js b/dist/node/chunks/config.js
index 2b39a20..deb26e6 100644
--- a/dist/node/chunks/config.js
+++ b/dist/node/chunks/config.js
@@ -30290,10 +30290,10 @@ const UrlRewritePostcssPlugin = (opts) => {
 		Once(root) {
 			const promises$2 = [];
 			root.walkDecls((declaration) => {
-				const importer = declaration.source?.input.file;
-				if (!importer) opts.logger.warnOnce("\nA PostCSS plugin did not pass the `from` option to `postcss.parse`. This may cause imported assets to be incorrectly transformed. If you've recently added a PostCSS plugin that raised this warning, please contact the package author to fix the issue.");
 				const isCssUrl = cssUrlRE.test(declaration.value);
 				const isCssImageSet = cssImageSetRE.test(declaration.value);
+				const importer = declaration.source?.input.file;
+				if (!importer && (isCssUrl || isCssImageSet)) opts.logger.warnOnce("\nA PostCSS plugin did not pass the `from` option to `postcss.parse`. This may cause imported assets to be incorrectly transformed. If you've recently added a PostCSS plugin that raised this warning, please contact the package author to fix the issue.");
 				if (isCssUrl || isCssImageSet) {
 					const replacerForDeclaration = async (rawUrl) => {
 						const [newUrl, resolvedId] = await opts.resolver(rawUrl, importer);
